ui/eureka/eureka.js

```diff
@@ -2256,7 +2293,7 @@ class EurekaTracker {
       } else {
         weatherStr += ' for ???';
       }
-      document.getElementById('label-weather0').innerHTML = weatherStr;
+      document.getElementById('label-weather0').innerHTML = this.strLocalization(weatherStr);
```

Comment:
Rather than doing localization by string replacement on strings that are appended together, could you localize the building of these strings, so that it's explicit?
```
let timeStrings = {
  weatherFor: {
    en: (weather, nowMs, stopTime) => {
      let weatherStr = gWeatherIcons[currentWeather];
      if (stopTime) {
        let min = (stopTime - nowMs) / 1000 / 60;
        weatherStr += ' for ' + Math.ceil(min) + 'm';
      } else {
        weatherStr += ' for ???';
      }
      return weatherStr;
    },
    ko: (weather, stopTime) => {
      // etc
    },
  },
  remainingTime: {
    en: (remainingMs) => {
    },
  },
},
```

That way if anybody ever changes the format of what it says in English then it won't break other languages.

Reply:
finished changing to this method.